File: Questão-6/Motor.go
package main

import (
	"fmt"
)

type Motor struct{}

func (m Motor) Ligar() {
	fmt.Println("Motor está Ligado")
}
func (m Motor) Desligar() {
	fmt.Println("Motor está Desligado")
}

type Carro struct {
	motor  Motor
	marca  string
	modelo string
}

func (c Carro) Ligar() {
	c.motor.Ligar()
	println("Carro está ligado")
}

func (c Carro) Desligar() {
	c.motor.Desligar()
	println("Carro está desligado")
}

func main() {
	carro := Carro{motor: Motor{}, marca: "Fiat", modelo: "Uno"}
	carro.Ligar()
	carro.Desligar()

}

File: Questão-6/Motor.py
class Motor:
  def ligar(self):
    print("Motor Ligado")
  def desligar(self):
    print("Motor Desligado")

class Carro:
  def __init__(self, marca, modelo):
    self.motor = Motor()
    self.marca = marca
    self.modelo = modelo

  def ligar(self):
    self.motor.ligar()
    print("Carro ligado")
  def desligar(self):
    self.motor.desligar()
    print("Carro Desligado")


carro = Carro("Fiat", "Uno")
carro.ligar()
carro.desligar()
